jokers/red_card.lua

```diff
@@ -0,0 +1,163 @@
+    add_to_deck = function(self, card, from_debuff)
+        if card.ability.extra.colour == "green" then
+            self.update_probabilities(card)
+        end
+    end,
+    remove_from_deck = function(self, card, from_debuff)
+        local old = card.ability.extra.colour
+        card.ability.extra.colour = "red"
+        card.ability.extra.triggers = card.ability.extra.triggers + 1
+        self.update_probabilities(card)
+        card.ability.extra.colour = old
+    end,
+    set_sprites = function(self, card, front)
+        if card.ability then
+            local colour = colours[card.ability.extra.colour]
+            card.children.center:set_sprite_pos(colour.pos)
+        end
+    end,
+    calculate = function(self, card, context)
+        if context.skipping_booster and not context.blueprint then
+            card.ability.extra.triggers = card.ability.extra.triggers + 1
+            self:switch_colour(card)
+            --red by default
+            local vars = { card.ability.extra.mult }
+            if card.ability.extra.colour == "blue" then
+                vars = { card.ability.extra.chips }
+            end
+            if card.ability.extra.colour == "yellow" then
+                vars = { card.ability.extra.money }
+            end
+            if card.ability.extra.colour == "green" then
+                vars = { card.ability.extra.probability }
+            end
+            return {
+                message = localize({
+                    type = "variable",
+                    key = "a_folly_" .. card.ability.extra.colour .. "_card",
+                    vars = vars,
+                }),
+                colour = colours[card.ability.extra.colour].message_colour,
+            }
+        end
+
+        if context.joker_main then
+            if card.ability.extra.colour == "red" then
+                return {
+                    mult = card.ability.extra.mult * card.ability.extra.triggers,
+                }
+            end
+            if card.ability.extra.colour == "blue" then
+                return {
+                    chips = card.ability.extra.chips * card.ability.extra.triggers,
+                }
+            end
+        end
+    end,
+    loc_vars = function(self, info_queue, card)
+        --red by default
+        local vars =
+            { card.ability.extra.mult, card.ability.extra.mult * card.ability.extra.triggers }
+        if card.ability.extra.colour == "blue" then
+            vars =
+                { card.ability.extra.chips, card.ability.extra.chips * card.ability.extra.triggers }
+        end
+        if card.ability.extra.colour == "yellow" then
+            vars =
+                { card.ability.extra.money * card.ability.extra.triggers, card.ability.extra.money }
+        end
+        if card.ability.extra.colour == "green" then
+            vars = {
+                card.ability.extra.probability * card.ability.extra.triggers,
+                card.ability.extra.probability,
```

Comment:
I think you should keep the ordering of the vars the same for all colors.
factor first and then the product, or opposite if you prefer.